spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/liquibase/LiquibaseProperties.java

```diff
@@ -114,6 +115,11 @@
 	 */
 	private boolean testRollbackOnUpdate;
 
+	/**
+	 * Name of the tag.
+	 */
+	private String tag;
```

Comment:
While reviewing this PR, I was trying to improve the documentation of that property and couldn't really find my way in the liquibase documentation. The original issue states:

> This property will help in generating roll back scripts for particular tags only when user has specified rollbackFile property.

Do you know more about this property and how it's actually used? I've looked into the liquibase source code and I am not sure I understood the scope of that property. If that's used only for rollbacks, I think we should make that clearer in the description of the property.

Reply:
when I was looking into the code I found `SpringLiquibase` uses `tag` property at `performUpdate` and `generateRollbackFile` methods. It means is executed during the update of changelogs or if additionally `rollbackFile` is also set will generate the file.

Reply:
Thanks for the feedback. Same but I think "Name of the tag" isn't enough for users to figure out what to do. Perhaps this is an issue in the Liquibase documentation?

Reply:
I poked around and I am confused. I don't think we should just have a "tag" property, I've [commented on the original issue](https://github.com/spring-projects/spring-boot/issues/19200#issuecomment-568689549).

Reply:
sorry for the late reply Stéphane! I have been thinking about it and TBH I think the workflow is pretty much deploying changes using the CLI and generating the rollback file from the application. As we have seen, the same property is used for rolling out and rollback. Said that, we should be very careful about the docs as you are worried about.

Delivering this feature, people will be able to:
1. Use `tag` property, apply the tag value to the scripts

or

2. Use `tag` property and `rollbackFile`,  generate the file to perform a rollback

Reply:
Hi - I am the Liquibase Community Manager and can comment some on the use cases, which seem to be a bit muddy in this ticket. 

I'm still learning about how Liquibase works in the Spring Boot ecosystem, so please bear with me if I get something wrong.

In Liquibase, there is a notion of a tag that can be applied to the DATABASECHANGELOG table (the table that Liquibase uses to keep track of which changeSets have been applied to a particuloar database). A tag is represented by adding a non-null value to the TAG column of the DATABASECHANGELOG table.

A tag can be applied in one of two ways - it can be included in the changelog, or it can be applied 'manually' by running the tag command, which is available in the CLI and as a maven goal. 

Once a tag has been applied, it can then be used by the rollback command. When used by the rollback command, any changesets that are represented in the DATABASECHANGELOG table that are after the tagged changeset will be rolled back. 

When rolling back, the operation can either be done 'live', or a SQL file can be generated containing the SQL commands needed. 

This is not well documented in the Liquibase docs, so I'll be updating those soon.

Reply:
@SteveDonie thanks for the feedback and improving the docs.

We need a one sentence that would be a good summary for the property and we had a hard time finding something succinct.

Here is a proposal

> Tag name to add to applied changes. Can be used with "rollbackFile" to generate a rollback script for all existing changes with that tag. 

Thoughts?

Reply:
I think that is pretty good. Maybe even call out that there are two different uses:

> Tag name to apply when applying database changes. Tag name to use as a target when rolling back already applied changes.

Reply:
When rolling back, you don't have to generate a rollbackFile. I'm not sure how it works in spring boot, but from the command line there are two commands - rollback, and rollbackSQL. One does the actual rollback, and the other generates the SQL to perform the rollback and shows it on the console (or sands it to a file).

Reply:
Thanks very much for the pointers, @SteveDonie.

Reply:
Indeed, thank you @stevedonie. FTR, Spring Boot "merely" instruments  `SpringLiquibase` which is part of Liquibase itself.